napari/utils/events/_tests/test_evented_model.py

```diff
@@ -461,6 +478,11 @@ def test_evented_model_with_property_setters():
     assert t.c == [2, 3]
     assert t.a == 2
     assert t.b == 3
+    t.d = 4
+    assert t.a == 2
+    assert t.b == 2
+    t.e = 100
+    assert t.a == 10
 
 
 def test_evented_model_with_property_setters_events():
```

Comment:
It may be nice to split this test on multiple functions (maybe with mocking in fixture). As it is long and test separate behaviors.

Reply:
I'm not sure how I could convert the mocking into a fixture; could you give me an example?

Reply:
```python
@pytest.fixture()
def mocked_object():
    t = T()
    t.events.a = Mock(t.events.a)
    t.events.b = Mock(t.events.b)
    t.events.c = Mock(t.events.c)
    t.events.d = Mock(t.events.d)
    t.events.e = Mock(t.events.e)
    return t

def test_evented_model_with_property_setters_events(mocked_object):
	assert 'c' in mocked_object.events # the setter has an event

    mocked_object.c = [10, 20]
    mocked_object.events.a.assert_called_with(value=10)
    mocked_object.events.b.assert_called_with(value=20)
    mocked_object.events.c.assert_called_with(value=[10, 20])
    mocked_object.events.d.assert_called_with(value=30)
    mocked_object.events.e.assert_called_with(value=100)

def test_evented_model_with_property_setters_events2(mocked_object):
    mocked_object.d = 8
    mocked_object.events.a.assert_called_with(value=4)
    mocked_object.events.b.assert_called_with(value=4)
    mocked_object.events.c.assert_called_with(value=[4, 4])
    mocked_object.events.d.assert_called_with(value=8)
    mocked_object.events.e.assert_called_with(value=40)
```

etc

Reply:
ok, easier than I thought :P

Reply:
I ended up making it into a parametrized test, I hope it's clear enough.